pkg/cloud/services/provider/provider.go

```diff
@@ -80,7 +80,6 @@ func NewClient(cloud clientconfig.Cloud, caCert []byte) (*gophercloud.ProviderCl
 	if cloud.AuthInfo != nil {
 		clientOpts.AuthInfo = cloud.AuthInfo
 		clientOpts.AuthType = cloud.AuthType
-		clientOpts.Cloud = cloud.Cloud
```

Comment:
what about a clouds.yaml contains multiple cloud ?

Reply:
I assume this does not matter. This is already handled in: https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/829/files#diff-dbf706fca1006fa16e85a98f355508c0891e685f846ff079d5ac1d6e6b308148L155

We're only passing one cloud (clientconfig.Cloud) into the NewClient func.

Reply:
Also the code path in AuthOptions never works when we set the cloud to a non empty string, because it always searches for a cloud.yaml file at the default locations which we don't have.

Reply:
@jichenjc wdyt, okay to merge?

Reply:
give me another day, I want to read it more (run out of time today)

Reply:
No problem, thx ;)

Reply:
we have a set of logics at line 52 / 70 above , such as 
`cloud, caCert, err = getCloudFromSecret(ctrlClient, namespace, openStackCluster.Spec.CloudsSecret.Name, openStackCluster.Spec.CloudName)`

so basically, we are providing info based on cluster.Spec.CloudName

if we don't want to honor this field, should we consider remove 
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/api/v1alpha3/openstackcluster_types.go#L38? 

actually we created a secret and this secret is a clouds.yaml and it might contains multiple clouds
I also think we don't want to (technically ?) support 1 CAPI + multiple openstack solution ? 
if that's the case, maybe we need reconsider adjust the whole `Cloud` related items and manage expecations of users

Reply:
Hm I we are using the CloudName here:
* https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/829/files#diff-dbf706fca1006fa16e85a98f355508c0891e685f846ff079d5ac1d6e6b308148R150-R154

We are even checking that it's not empty here:
* https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/829/files#diff-dbf706fca1006fa16e85a98f355508c0891e685f846ff079d5ac1d6e6b308148R125-R127

So in my opinion we need the property and it does exactly what we want. It picks the right cloud if we have multiple in the cloud.yaml. The only problem is that we don't need it in `clientconfig.AuthOptions`. `clientconfig.AuthOptions` is not the func which parses the cloud.yaml so it doesn't need the cloud. 

I don\t really understand what you mean, but we cannot drop the property. We need it, just not here.

Reply:
(Also cloud.Cloud is not the **same** as the cloudName in our spec)

Reply:
ok, you are right, I was confused by cloud and cloudName variable at first then I misunderstand the logic 
after further read I think you are right , sorry for the confusion

Reply:
No problem, it's confusing :). I also would like to use not that much of the AuthOptions func unfortunately the only way to do this is to copy it :/